packages/big-design/src/components/Switch/styled.tsx

```diff
@@ -0,0 +1,91 @@
+import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
+import { hideVisually } from 'polished';
+import styled, { css } from 'styled-components';
+
+import { withTransition } from '../../mixins/transitions';
+
+export const HiddenCheckbox = styled.input`
+  ${hideVisually()}
+`;
+
+type SwitchLabelProps = {
+  checked?: boolean;
+  disabled?: boolean;
+};
+
+const getShadow = (checked?: boolean) =>
+  `0px 0px 0px 4px ${checked ? 'rgba(62, 103, 248, 0.2)' : 'rgba(140, 147, 173, 0.2)'};`;
```

Comment:
Couple of things I'm not a fan about this. *A)* This should be done inline and it's fine if it's not DRY. Styled Components should optimize it appropriately. *B)* We should be using the theme object instead of hand writing the colors. I'll add a suggestion below how it probably should be.

Reply:
Yeah, I am not a fan either. This was me trying to achieve what the final outcome was. The better solution would be using `theme` as outlined below.